lib/facter/custom_facts/core/execution/base.rb

```diff
@@ -100,18 +100,22 @@ def execute_command(command, on_fail, logger = nil, time_limit = nil)
             @log.debug("Executing command: #{command}")
             out, stderr = Open3.popen3(opts, command.to_s) do |_, stdout, stderr, wait_thr|
               pid = wait_thr.pid
-              output = +''
-              err = +''
+              stdout_messages = +''
+              stderr_messages = +''
```

Comment:
why not use a simple initialize?
```ruby
stdout_messages = ''
stderr_messages = ''
```

Reply:
they will be treated as frozen strings and this will generate an error

Reply:
I've started using `String.new` to make it clearer that I'm creating a string that isn't frozen

Reply:
Done

Reply:
Unfortunately i get the following error when i try to use `String.new` : `lib/facter/custom_facts/core/execution/base.rb:104:33: C: Performance/UnfreezeString: Use unary plus to get an unfrozen string literal.`